### ISSUE TITLE
show alternative package if it doesn't support Python 3

### ISSUE THREAD

**Issue by chhantyal** [chhantyal/py3readiness#9]
Just a thought before going ahead: 
Would be nice to provide an alternative package if
1. Package is abandoned and has drop-in replacement
2. Package is not going to be supporting Python 3 because there already alternative package doing so.

etc.

Color would be yellow?

**Comment by ztane** [comment]
+1 for this. For example it is not immediately obvious that sqlalchemy-migrate is a third-party effort and the preferred migration tool for sqlalchemy is the package named alembic.

**Comment by robertlagrant** [comment]
Yeah it should still be green, but with a note that a drop-in replacement exists?

**Comment by nicktimko** [comment]
Point of fact, MySQLdb is basically dead, but there's now a fork (drop-in replacement, no API changes) recommended by Django in their docs:

> ## [MySQL DB API Drivers](https://docs.djangoproject.com/en/1.8/ref/databases/#mysql-db-api-drivers)
> * [MySQLdb](https://pypi.python.org/pypi/MySQL-python/1.2.4) is a native driver that has been developed and supported for over a decade by Andy Dustman.
> * [mysqlclient](https://pypi.python.org/pypi/mysqlclient) is a fork of **MySQLdb** which notably supports Python 3 and can be used as a drop-in replacement for MySQLdb. At the time of this writing, this is **the recommended choice** for using MySQL with Django.

**Comment by robertlagrant** [comment]
[PyMySQL](https://github.com/PyMySQL/PyMySQL) also seems to work nicely - pure-Python replacement that doesn't have a compile step on install. Nice for us fortunate enough to have to use Windows :)

**Comment by mathiasertl** [comment]
+1 from me.

(I'd like to add that I've made a 2.7/3.4+ fork for Fabric: https://pypi.python.org/pypi/Fabric3)

**Comment by brettcannon** [comment]
I should mention that some of the overrides in caniusepython3 are actually drop-in alternatives so some of that data is already collected for you (e.g., look at how BeautifulSoup and PIL are overridden).

**Comment by mathiasertl** [comment]
@brettcannon in that case can you add [Fabric3](https://pypi.python.org/pypi/Fabric3) as drop-in alternative to Fabric?

**Comment by brettcannon** [comment]
@mathiasertl Please file an issue or send a PR against brettcannon/caniusepython3

**Comment by nickjacobson** [comment]
Another example is https://pypi.python.org/pypi/ipaddr which was forked and became a builtin library in Python 3: https://docs.python.org/3/library/ipaddress.html

**Comment by nicktimko** [comment]
The wheel of win shows more and more win, but it's perhaps a bit strange when the Python 3 alternatives "overrides" aren't signified in any way. Noting which have alternate packages via https://github.com/brettcannon/caniusepython3/blob/master/caniusepython3/overrides.json might be prudent so there isn't any confusion if someone tries to install a package name (e.g. `mysql-python`) that's green but it then fails under 3.

**Comment by nickjacobson** [comment]
2 more that are in white, and that have a version in Python 3:
"python-openid": "https://pypi.python.org/pypi/python3-openid" (already in overrides.json)
"python-ldap": "https://github.com/pyldap/pyldap" (I requested to have this added to overrides.json)

**Comment by mcgeeco** [comment]
Sorry if I've misunderstood what's supposed to be happening here, but why is Fabric listed as green with no mention on the site that you have to use a fork?

**Comment by chhantyal** [comment]
@mcgeeco Sorry if it was misleading. I clarified the text here https://github.com/chhantyal/py3readiness/commit/1aad0802dde16819f1be300a3548685d4669a3fe will be updated in next deployment

**Comment by mcgeeco** [comment]
Ah that makes sense. Personally I like the yellow idea. Thanks!
